### PR TITLE
Fix Barrel HCAL Volume ID's To Allow Proper Clustering

### PR DESCRIPTION
**Pull request by johnlajoie**
### Briefly, what does this PR introduce?

This PR updates the volume ID's for the various detector elements (sector, tower, tile) and corresponding readout configuration. The "barrel" identifier has been removed.  The fixed volume ID's now allow for proper clustering, which requires an additional fix to EICrecon which will be committed by Derek Anderson. 

As a side benefit, this also includes simplified GDML solids for the tiles. This work was done last fall but never merged in the rush to the first simulation campaign. 

### What kind of change does this PR introduce?
- [X] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [X] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR change default behavior?

With the old volume ID configuration and EICrecon configuration we were actually merging all the tiles in a sector, not all the tiles in a tower, to create the merged hits.

 
hitvsclustnum varyinghitmergefields_baseline e10th70n10kpip d16m2y2023_720

This first plot shows the old configuration - notice that the number of merged hits and number of clusters is very similar, and this persisted across all energies.  This was a symptom that the merging was too broad and we only saw a difference when a cluster split across a sector boundary. 

hitvsclustnum varyinghitmergefields_onlytilesinhitmerger e10th70n10kpip d16m2y2023_720

This plot shows the new configuration, with towers energies created from merged tiles.  As you can see there are many more towers with energy that clusters (since clusters should be a collection of towers). 

Many thanks to Dmitry Kalinkin for his help with getting the volume ID's straightened out!

### CODE REVIEW

File: src/BarrelHCalCalorimeter_geo.cpp [hunk 1]
```diff
@@ -531,7 +531,7 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
         ChimneySector, i + 1,
         Transform3D(RotationZ(((i - 1) * 2 * M_PI / 32) + tweak_sectors[i + 1]), Translation3D(0, 0, 0)));
     sect_phv.addPhysVolID("system", det_id);
-    sect_phv.addPhysVolID("barrel", 0);
+    //sect_phv.addPhysVolID("barrel", 0);
```
Comment (veprbl) [inline src/BarrelHCalCalorimeter_geo.cpp:534]:
```suggestion
```

**Comment by veprbl** [comment]
> As a side benefit, this also includes simplified GDML solids for the tiles. This work was done last fall but never merged in the rush to the first simulation campaign.

That wasn't missed, it seems: https://github.com/eic/epic/pull/266

File: src/BarrelHCalCalorimeter_geo.cpp [hunk 2]
```diff
@@ -546,7 +546,7 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
                              Transform3D(RotationZ((-2.075 * M_PI / 32) + (i - 3) * (2 * M_PI / 32) + tweak_sectors[i]),
                                          Translation3D(0, 0, 0)));
     sect_phv.addPhysVolID("system", det_id);
-    sect_phv.addPhysVolID("barrel", 0);
+    //sect_phv.addPhysVolID("barrel", 0);
```
[the remark this answers is not in the document]
Comment (johnlajoie) [reply src/BarrelHCalCalorimeter_geo.cpp:549]: Fine.

File: src/BarrelHCalCalorimeter_geo.cpp (hunk 1 above)
Comment (johnlajoie) [reply src/BarrelHCalCalorimeter_geo.cpp:534]: Fine

File: src/BarrelHCalCalorimeter_geo.cpp [hunk 3]
```diff
@@ -63,7 +63,7 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
 
   PlacedVolume env_phv = motherVol.placeVolume(envelope);
   env_phv.addPhysVolID("system", det_id);
-  env_phv.addPhysVolID("barrel", 0);
+  //env_phv.addPhysVolID("barrel", 0);
```
[the remark this answers is not in the document]
Comment (johnlajoie) [reply src/BarrelHCalCalorimeter_geo.cpp:66]: Also fine

**Comment by johnlajoie** [comment]
> 

No, #266 dealt with removing unneeded small details in the new tile shapes.  After that I was also able to reduce the overall number of faces used in the tile description (that are flat shapes after all).  This should make things easier of the geometry subsystem when dealing with the tessellated volumes.   The new xml file in this pull request included the simplified tile geometry with a smaller number of faces.

File: src/BarrelHCalCalorimeter_geo.cpp [hunk 4]
```diff
@@ -515,7 +515,7 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
         Tower[i], i + 48,
         Transform3D(RotationZ(tileRotateStart - octileRotateStart + 5 * (360.0 / 320.0) * dd4hep::deg + tweak_tiles[i]),
                     Translation3D(0.0, 0.0, 0.0)));
-    tower_phv1.addPhysVolID("tower", i + 72);
+    tower_phv1.addPhysVolID("tower", i+24); // upper phi
```
[the remark this answers is not in the document]
Comment (johnlajoie) [reply src/BarrelHCalCalorimeter_geo.cpp:517]: In principle yes, but it's also pretty straightforward to take the mod 24 to get the tower number in the second row.  For now this is what Derek's new code does, and in the future we will eliminate the tower summing entirely in favor of clustering based on the individual scintillators.  For now I would prefer not to add to the code that we will cut an simplify soon in any case.

**Comment by veprbl** [comment]
> No, #266 dealt with removing unneeded small details in the new tile shapes. After that I was also able to reduce the overall number of faces used in the tile description (that are flat shapes after all). This should make things easier of the geometry subsystem when dealing with the tessellated volumes. The new xml file in this pull request included the simplified tile geometry with a smaller number of faces.

There is nothing on that seen in the diff: https://github.com/eic/epic/pull/367/files

**Review by veprbl — APPROVED** [review]

**Comment by veprbl** [comment]
@wdconinc Sorry to bother, but the eicweb jobs appear stuck. Could you check?

**Comment by wdconinc** [comment]
> @wdconinc Sorry to bother, but the eicweb jobs appear stuck. Could you check?

The jobs will always fail from forks. We don't have a way (yet) to accept CI jobs for PRs from forks.

**Comment by wdconinc** [comment]
OT: Some ideas are:
- accept running jobs from forks when user is member of eic group,
- setup eic bot that can trigger full pipeline with a token that has access.

**Comment by wdconinc** [comment]
For now, please work in a branch in this repo.

**Comment by veprbl** [comment]
Can we make an exception this time and merge

**Comment by wdconinc** [comment]
Has eicrecon already been fixed to not use the `barrel` field? Has someone tested this branch against eicrecon:main?

**Comment by wdconinc** [comment]
I guess that's what https://github.com/eic/EICrecon/pull/498 did.

**Comment by veprbl** [comment]
There is nothing in eicrecon to fix for this change.

**Comment by wdconinc** [comment]
@johnlajoie don't develop in your `main` branch :-)

**Pull request merged**